threesixty/views.py

```diff
@@ -260,21 +262,39 @@ def post(self, request, *args, **kwargs):
 
     def get(self, request, *args, **kwargs):
         self.participant = get_object_or_404(models.Participant, email=self.email, survey=self.survey)
+        if self.kwargs.get('question_pk'):
+            return self.get_specific_question(request, args, kwargs)
+        else:
+            return self.get_random_question(request, args, kwargs)
+
+    def get_specific_question(self, request, args, kwargs):
+        question_pk = self.kwargs['question_pk']
```

Comment:
You are getting it for the second time from the kwargs. Why not attach it to the instance in the get method.

Reply:
I would not attach it to the instance, but just get it once.